### PR TITLE
fix autograd bug re: inclusion extraneous field mnts

### PR DESCRIPTION
**Pull request by tylerflex**
still not sure:
this isn't triggered by regular autograd code. might be something with the metrics

### CODE REVIEW

**Review by yaugenst-flex — APPROVED** [review]

File: tidy3d/components/data/monitor_data.py [hunk 1]
```diff
@@ -1093,6 +1093,10 @@ def to_adjoint_field_sources(self, fwidth: float) -> List[CustomCurrentSource]:
                 if not np.all(values == 0):
                     src_field_components[name] = ScalarFieldDataArray(values, coords=coords)
 
+            # dont include this source if no data
+            if all(fld_cmp is None for _, fld_cmp in src_field_components.items()):
```
Comment (yaugenst-flex) [inline tidy3d/components/data/monitor_data.py]: `src_field_components.values()`?

**Review by yaugenst-flex — APPROVED** [review]

File: tidy3d/components/data/monitor_data.py (hunk 1 above)
Comment (tylerflex) [reply tidy3d/components/data/monitor_data.py]: I think pylint used to complain about iterating over values, so I got in the habit of always doing over `items()`. but not really a strong preference.

Comment (yaugenst-flex) [reply tidy3d/components/data/monitor_data.py]: hm I don't see a reason for this. might have been an artifact of possibly unordered dicts, but that's not the case anymore as of python 3.7..

Comment (yaugenst-flex) [reply tidy3d/components/data/monitor_data.py]: e.g. ruff will complain by default about iterating over items if only keys or values are used. we might have disabled that rule 🤔

**Comment by tylerflex** [comment]
fixed values() iteration but still am not able to reproduce this error. (basically the test I added passes even without the change) would be good (but not super high priority) to understand why exactly this popped up and reproduce it in a test

**Comment by yaugenst-flex** [comment]
yeah I also can't reproduce it. I think it's because emulated run does not exactly reproduce what is happening in regular run, because the code will fail is using regular run

**Comment by momchil-flex** [comment]
Merging but this would need further investigation

**Pull request merged**